imap_data_access/file_naming.py

```diff
@@ -0,0 +1,33 @@
+"""Methods for managing and validating filenames."""
+import re
+
+def extract_filename_components(filename: str):
+    """
+    Extracts all components from filename.
+
+    Parameters
+    ----------
+    filename : str
+        Path of dependency data.
+
+    Returns
+    -------
+    components : dict
+        Dictionary containing components.
+
+    """
+    pattern = (
+        r"^imap_"
+        r"(?P<instrument>[^_]+)_"
+        r"(?P<datalevel>[^_]+)_"
+        r"(?P<descriptor>[^_]*)_?"  # optional
+        r"(?P<startdate>\d{8})_"
+        r"(?P<enddate>\d{8})_"
+        r"(?P<version>v\d{2}-\d{2})"
+        r"\.(cdf|pkts)$"
+    )
+    match = re.match(pattern, filename)
+    if match is None:
+        return
```

Comment:
Do we want to raise an error here instead indicating its a bad filename? I think path_helper.py might have more checks and put some helpful messages in if it doesn't match the structure, do we want to bring those over as well?